drivers/adc/adaq8092/adaq8092.c

```diff
@@ -0,0 +1,683 @@
+	ret = adaq8092_write(dev,  ADAQ8092_REG_RESET, field_prep(ADAQ8092_RESET, 1));
+	if (ret)
+		goto error_par_ser;
+
+	mdelay(100);
+
+	/* Device Initialization */
+	ret = adaq8092_set_pd_mode(dev, init_param.pd_mode);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_clk_pol_mode(dev, init_param.clk_pol_mode);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_clk_phase_mode(dev, init_param.clk_phase_mode);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_clk_dc_mode(dev, init_param.clk_dc_mode);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_lvds_cur_mode(dev, init_param.lvds_cur_mode);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_lvds_term_mode(dev, init_param.lvds_term_mode);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_dout_en(dev, init_param.dout_en);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_dout_mode(dev, init_param.dout_mode);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_test_mode(dev, init_param.test_mode);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_alt_pol_en(dev, init_param.alt_bit_pol_en);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_data_rand_en(dev, init_param.data_rand_en);
+	if (ret)
+		goto error_par_ser;
+
+	ret = adaq8092_set_twos_comp(dev, init_param.twos_comp);
+	if (ret)
+		goto error_par_ser;
+
+	*device = dev;
+
+	return 0;
+
+error_par_ser:
+	gpio_remove(dev->gpio_par_ser);
+error_en_1p8:
+	gpio_remove(dev->gpio_en_1p8);
+error_adc_pd2:
+	gpio_remove(dev->gpio_adc_pd2);
+error_adc_pd1:
+	gpio_remove(dev->gpio_adc_pd1);
+error_spi:
+	spi_remove(dev->spi_desc);
+error_dev:
+	free(dev);
+
+	return ret;
+}
+
+/**
+ * @brief Remove the device and release resources.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adaq8092_remove(struct adaq8092_dev *dev)
+{
+	int ret;
+
+	ret = spi_remove(dev->spi_desc);
+	if (ret)
+		return ret;
+
+	ret = gpio_remove(dev->gpio_adc_pd1);
+	if (ret)
+		return ret;
+
+	ret = gpio_remove(dev->gpio_adc_pd2);
+	if (ret)
+		return ret;
+
+	ret = gpio_remove(dev->gpio_en_1p8);
+	if (ret)
+		return ret;
+
+	ret = gpio_remove(dev->gpio_par_ser);
+	if (ret)
+		return ret;
+
+	free(dev);
+
+	return ret;
+}
+
+/**
+ * @brief Set the device powerodown mode.
+ * @param dev - The device structure.
+ * @param mode - The device mode.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adaq8092_set_pd_mode(struct adaq8092_dev *dev,
+			 enum adaq8092_powerdown_modes mode)
+{
+	int ret;
+
+	ret = adaq8092_update_bits(dev, ADAQ8092_REG_POWERDOWN,
+				   ADAQ8092_POWERDOWN_MODE,
+				   field_prep(ADAQ8092_POWERDOWN_MODE, mode));
+	if (ret)
+		return ret;
+
+	dev->pd_mode = mode;
+
+	return 0;
+}
+
+/**
+ * @brief Get the device powerdown mode.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adaq8092_get_pd_mode(struct adaq8092_dev *dev)
```

Comment:
nitpick: should we return corresponding `enum` instead of just `int`?

Reply:
ACK